lib/portlayer/exec/container.go

```diff
@@ -151,6 +151,20 @@ func (s State) String() string {
 	return ""
 }
 
+// CurrentState returns current state.
+func (c *Container) CurrentState() State {
+	c.m.Lock()
+	defer c.m.Unlock()
+	return c.state
+}
+
+// SetState changes container state.
+func (c *Container) SetState(s State) {
```

Comment:
Also `Test & Set` or we still have a race - see the `state = existingState` path.